### ISSUE TITLE
User Tasks and "Draft submissions"

### ISSUE THREAD

**Issue by StephenOTT** [kiegroup/kogito-runtimes#1909]
### Description

Is there any concepts or plans for User Task draft submissions?  Where a user can Save As Draft their data submission.  When a user loads the task later it would load that saved draft of data.

This could be something as simple as a processKey/instanceId/taskName/instanceId/draft
which would allow data draft submission against the schema / same setup as the submission object.

Thanks!

**Comment by fjtirado** [comment]
No, we do not have such concept, but why not to use the existing /{id}/{taskName/taskId}/{taskInstanceId} PUT API to change the data without changing task phase (the POST call) to emulate that. Basically saving as draft is to change the task model without changing the current phase. That can be done with the mentioned PUT api.

**Comment by StephenOTT** [comment]
@fjtirado I see that as a destructive change, where a draft should be viewed as a non-destructive change: something that can be undone, abandoned, and not necessarily destroying the original state for other users (users who did not create the draft).  If we make a PUT on the UT, it would change the underlying data in the model (whether it be local or process level data).
